src/main/java/org/folio/event/handler/ItemCreateAsyncRecordHandler.java

```diff
@@ -96,4 +122,101 @@ private void updatePieceFields(List<Piece> pieces, String holdingId, String tena
       }
     });
   }
+
+  // Find affected POLs from updated pieces, find all other POL pieces, reconstruct POL locations from all pieces
+  private Future<Void> processPoLinesUpdate(List<Piece> pieces, JsonObject itemObject, String tenantIdFromEvent,
+                                            String centralTenantId, Map<String, String> headers, Conn conn) {
+    if (CollectionUtils.isEmpty(pieces)) {
+      log.info("processPoLinesUpdate:: No updated pieces were found to update for item: '{}' and tenant: '{}' in centralTenant: {}",
+        itemObject.getString(ID.getValue()), tenantIdFromEvent, centralTenantId);
+      return Future.succeededFuture();
+    }
+    var poLineIds = pieces.stream()
+      .map(Piece::getPoLineId)
+      .distinct()
+      .toList();
+    log.info("processPoLinesUpdate:: Preparing '{}' poLineIds for update processing", poLineIds.size());
+    return poLinesService.getPoLinesByLineIdsByChunks(poLineIds, conn)
+      .compose(poLines -> {
+        var poLinePiecePairsFutures = new ArrayList<Future<Pair<PoLine, List<Piece>>>>();
+        poLines.forEach(poLine -> {
+          var piecesFuture = pieceService.getPiecesByPoLineId(poLine.getId(), conn)
+            .map(allPieces -> Pair.of(poLine, allPieces));
+          poLinePiecePairsFutures.add(piecesFuture);
+        });
+        return collectResultsOnSuccess(poLinePiecePairsFutures);
+      })
+      .compose(poLinePiecePairs -> updatePoLines(poLinePiecePairs, itemObject, centralTenantId, headers, conn))
+      .mapEmpty();
+  }
+
+  // Update locations and searchLocationIds for each POL
+  private Future<List<PoLine>> updatePoLines(List<Pair<PoLine, List<Piece>>> poLinePiecePairs, JsonObject itemObject,
+                                             String centralTenantId, Map<String, String> headers, Conn conn) {
+    var updatedPoLines = new ArrayList<PoLine>();
+    log.info("updatePoLines:: Updating '{}' POL(s) for item: '{}' in centralTenant: '{}'",
+      poLinePiecePairs.size(), itemObject.getString(ID.getValue()), centralTenantId);
+    poLinePiecePairs.forEach(poLineListPair -> {
+      var poLine = poLineListPair.getLeft();
+      var pieces = poLineListPair.getRight();
+      var locationsUpdated = updatePoLineLocations(pieces, poLine);
+      var searchLocationIdsUpdated = updatePoLineSearchLocationIds(itemObject, poLine);
```

Comment:
we can change logs of these two method info to debug because we have not found logs in 169 line and I saw Json.encode() usage